sympy/solvers/tests/test_solveset.py

```diff
@@ -2172,12 +2172,12 @@ def test_invert_modular():
     assert invert_modular(Mod(x + 8, 7), S(5), n, x) == \
             (x, ImageSet(Lambda(n, 7*n + 4), S.Integers))
     assert invert_modular(Mod(x**2 + x, 7), S(5), n, x) == \
-            (Mod(x**2 + x, 7), 5)
+            (x, ImageSet(Lambda(n, 7*n + 3), S.Integers))
```

Comment:
Let us take this example According to your code 
`7*n+3` is **all the solutions* to `Mod(x**2 + x, 7)-5`.
But that's not the case :
Clearly solution of `x**2 + x - 5=0` gives one of the solution to this equation.
i.e `x = (-1+/- sqrt(21))/2` which is not included in your solution.
I hope you understand the crux of the problem. Try to be creative to solve this problem. Its nice you are working on this. Try to add it in solve_modular.

Reply:
Yes you are right, i get it now. I should implement it to `solve_modular` function which is designed to return `set of integer solutions`. 
It would be very difficult to invert a general polynomial modular equation to general solutions, but it is relatively easier to invert it to only integer solutions.

Reply:
One more thing, the current implementation of invert_modular does not seem to give all solution.
For example: `invert_modular(Mod(x**4, 7), S(5), n, x)` output `(x, EmptySet)` . But 5**0.25 is a solution to `Mod(x**4,7) -5`.

Reply:
Yeah this bug was irritating while we were creating `_invert_modular`. Can you try to fix this? Although this is in the further development task of `solve_modular`.

Reply:
@smichr @jmig5776 I don't think that this bug can be solved for a general polynomial. For example:  `mod(x**2+x-5, 10) =0`, integer solution can be easily calculated but for other solution we have to solve for roots of `x**2+x-5 = 10k` where k is any integer. For f(x)=x**n it may be done but for general polynomial this will not be feasible as there will be no relation between the roots for different k. Moreover `_invert_modular` function is only used by `_solve_modular`, and the return type of `_solve_modular` is set of integers solutions so this is suitable for `_solve_modular` but not for `_invert_modular`.
For now the changes i have made should do the trick for solving polynomial congruence equation. What are your thoughts on this?